File: clinic-core/src/main/java/com/nonier/cliniccore/entity/Doctor.java
package com.nonier.cliniccore.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(exclude = "doctorSpecializations")
@ToString(exclude = "doctorSpecializations")
@Table(name = "doctor")
public class Doctor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(name = "age_group", nullable = false)
    private AgeGroup ageGroup;

    @Column(name = "work_experience")
    private Integer workExperience;

    @Enumerated(EnumType.STRING)
    @Column(name = "rank", nullable = false)
    private Rank rank;

    @OneToMany(mappedBy = "doctor", orphanRemoval = true)
    private List<DoctorSpecialization> doctorSpecializations = new ArrayList<>();

    @OneToMany(mappedBy = "doctor", orphanRemoval = true)
    private List<Consultation> consultations = new ArrayList<>();

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;
}

File: clinic-frontend/src/app/model/Doctor.ts
import {AgeGroup} from "./AgeGroup";
import {Rank} from "./Rank";
import {User} from "./User";
import {Specialization} from "./Specialization";
import {Consultation} from "./Consultation";

export class Doctor {
  id: number;
  ageGroup: AgeGroup;
  workExperience: number;
  rank: Rank;
  user: User;
  specializations: Specialization[];
  consultations: Consultation[];

  constructor(id: number, ageGroup: AgeGroup, workExperience: number, rank: Rank, user: User,
              specializations: Specialization[], consultations: Consultation[]) {
    this.id = id;
    this.ageGroup = ageGroup;
    this.workExperience = workExperience;
    this.rank = rank;
    this.user = user;
    this.specializations = specializations;
    this.consultations = consultations;
  }
}
